src/Common/Core/Impl/Extensions/IOExtensions.cs

```diff
@@ -84,5 +84,16 @@ public static class IOExtensions {
             }
             return path.Length;
         }
+
+        public static string EnsureTrailingSlash(this string path) {
+            if (!string.IsNullOrEmpty(path)) {
+                char slash = path.IndexOf(Path.AltDirectorySeparatorChar) >= 0 ? Path.AltDirectorySeparatorChar : Path.DirectorySeparatorChar;
+                if (path[path.Length - 1] != slash) {
```

Comment:
I think here we still need to check for both normal and alt, otherwise we may end up with something like `C:\b/a\/`